crates/noirc_evaluator/src/ssa/anchor.rs

```diff
@@ -250,9 +250,15 @@ impl Anchor {
                             return Some(CseAction::ReplaceWith(a));
                         }
                     }
-                    Operation::Store { index, value, .. } => {
+                    Operation::Store { index, value, predicate, .. } => {
                         if !ctx.maybe_distinct(*index, b_idx) {
-                            return Some(CseAction::ReplaceWith(*value));
+                            if ctx.is_one(crate::ssa::conditional::DecisionTree::unwrap_predicate(
+                                ctx, predicate,
```

Comment:
I think we should import this function, DecisionTree, or have it be a method on ctx